### PR TITLE
_yaml method for stac obj

### PR DESCRIPTION
**Pull request by jukent**
Per NASA-ACCESS work with @jhamman

This PR makes it so `.yaml()` will accept a stac-obj and create a YAML that includes all of the data for its assets.

Recreation of #103 in a safe branch.

One file (`sandbox.ipynb`) will be removed before push.

### CODE REVIEW

**Comment by jukent** [comment]
@jhamman Can you approve the workflow?

**Comment by jhamman** [comment]
@scottyhq and @martindurant - if you have a minute to take a look here, your review would be appreciated. To review, the goal here is to enabling serialization of intake-stac objects. Things got complicated because intake-stac maintains a reference to underlying stac objects which, requires some careful packing/unpacking.

File: intake_stac/catalog.py [hunk 1]
```diff
@@ -354,7 +363,7 @@ def __init__(self, key, asset):
             name=key,
             description=asset.title,
             driver=driver,
-            direct_access=True,
+            direct_access='allow',
```
Comment (martindurant) [inline intake_stac/catalog.py:366]: In hindsight, this is far from obvious! It won't matter to anyone not running an Intake server.

File: intake_stac/catalog.py [hunk 2]
```diff
@@ -333,6 +333,15 @@ def stack_bands(self, bands, path_as_pattern=None, concat_dim='band'):
 
         return CombinedAssets(configDict)
 
+    def _yaml(self):
+        data = {'metadata': {}, 'sources': {}}
+        data['metadata'].update(self.metadata)
+        for key, source in self.items():
+            data['sources'][key] = source._yaml()['sources']['stac_asset']
```
Comment (martindurant) [inline intake_stac/catalog.py:340]: Effectively, you recreate the stac object from this string unique identifier? If I've understood it correctly, then everything seems in order here.

Comment (jukent) [reply intake_stac/catalog.py:340]: Yes exactly

**Comment by jhamman** [comment]
Thanks again for working on this @jukent!

**Pull request merged**